docs/topics/spiders.rst

```diff
@@ -145,6 +145,41 @@ scrapy.Spider
        :param kwargs: keyword arguments passed to the :meth:`__init__` method
        :type kwargs: dict
 
+   .. classmethod:: update_settings(settings)
+
+       The ``update_settings()`` method is used to modify the spider's settings
+       and is called during initialization of a spider instance.
+
+       It takes a :class:`~scrapy.settings.Settings` object as a parameter and
+       can add or update the spider's configuration values. This method is a class method,
+       meaning that it is called on the :class:`~scrapy.Spider` class and allows all instances
+       of the spider to share the same configuration.
+
+       One of the main advantages of ``update_settings()`` is that it allows
+       you to dynamically add, remove or change settings based on other settings 
+       or other external factors.
+
+       For example, suppose a spider needs to modify :setting:`FEEDS`:
+
+       .. code-block:: python
+
+           import scrapy
+
+
+           class MySpider(scrapy.Spider):
+               name = "myspider"
+               custom_feed = {
+                   "/home/user/documents/items.json": {
+                       "format": "json",
+                       "indent": 4,
+                   }
+               }
+
+               @classmethod
+               def update_settings(cls, settings):
+                   settings.setdefault("FEEDS", {}).update(cls.custom_feed)
+                   super().update_settings(settings)
```

Comment:
Just want to confirm that wouldn't call to the `super().update_settings` be the first thing to do and then override the required settings here.

Reply:
I think that’s what you would usually want, yes, but not necessarily. For example, if you define `update_settings` in a base spider class, and a child spider class defined `custom_setting`, you may want that `custom_settings` to be able to override what the base class does in `update_settings`, in which case the code here is what you want.